simplq/src/main/java/me/simplq/service/TokenService.java

```diff
@@ -99,6 +99,9 @@ public TokenDetailResponse createToken(CreateTokenRequest createTokenRequest) {
                     throw SQInvalidRequestException.queueDeletedException();
                   } else if (queue.isFull()) {
                     throw SQInvalidRequestException.queueIsFullException();
+                  } else if (queue.isSelfJoinAllowed() || !loggedInUserInfo.getUserId()
```

Comment:
This should be extracted into other methods and use function composition. It is very hard to follow.
```java
  Predicate<Queue> isNotPaused() {
    return queue -> {
      if (queue.getStatus() == QueueStatus.PAUSED) {
        throw SQInvalidRequestException.queuePausedException();
      }

      return true;
    };
  }

  Predicate<Queue> isNotDeleted() {
    return queue -> {
      if (queue.getStatus() == QueueStatus.DELETED) {
        throw SQInvalidRequestException.queueDeletedException();
      }

      return true;
    };
  }

  Predicate<Queue> isNotFull() {
    return queue -> {
      if (queue.isFull()) {
        throw SQInvalidRequestException.queueDeletedException();
      }

      return true;
    };
  }

  void validateQueue(Queue queue) {
    isNotPaused()
        .and(isNotDeleted())
        .and(isNotFull())
        .test(queue);
  }
```
and then just call ```validateQueue(queue)```

Reply:
Also we can extract validations in an validation class and have the code in one place.

Reply:
lovely. Thanks. Will do this tomorrow morning.

Reply:
@chalx I will do this as a separate PR.